packages/components/src/Drawer/Drawer.component.js

```diff
@@ -159,15 +164,33 @@ DrawerFooter.propTypes = {
 	children: PropTypes.node,
 };
 
-export function combinedFooterActions(onCancelAction, footerActions) {
-	if (!onCancelAction) {
-		return footerActions;
-	}
+export function combinedFooterActions(onCancelAction, footerActions, activeTabItem = {}) {
+	const onCancelItem =
+		(onCancelAction && onCancelAction.position === ON_CANCEL_ACTION_POSITION_FOOTER) ||
+		(onCancelAction && !onCancelAction.position)
+			? onCancelAction
+			: null;
 	const enhancedFooterActions = Object.assign({}, footerActions);
 	enhancedFooterActions.actions = {
 		...enhancedFooterActions.actions,
-		left: [...get(enhancedFooterActions, 'actions.left', []), onCancelAction],
+		left: [
+			...get(enhancedFooterActions, 'actions.left', []),
+			...get(activeTabItem, 'actions.left', []),
+		],
+		center: [
+			...get(enhancedFooterActions, 'actions.center', []),
+			...get(activeTabItem, 'actions.center', []),
+		],
+		right: [
+			...get(enhancedFooterActions, 'actions.right', []),
+			...get(activeTabItem, 'actions.right', []),
+		],
 	};
+
+	if (onCancelItem) {
```

Comment:
Is it normal that if I add a `onCancelAction` whereas some `actions.left` are already defined, the `cancel` one sticks on the right of the actuals `actions.left` ?

<img width="495" alt="image" src="https://user-images.githubusercontent.com/26482371/41365297-51c1704e-6f39-11e8-84df-821c4f69677a.png">

Reply:
i think it is because there are lot of actions and the space dedicated for each button is not sufficient.
Normally, in each part, there is a space between actions but this space desappears when there are lot of action